index.html

```diff
@@ -703,15 +703,17 @@ <h2 class='informative'>Implementation Patterns</h2>
     </section>
 
     <section id="messaging-spi">
-      <h2>Messaging SPI</h2>
```

Comment:
I think (and there seemed to be support for the idea) that the use of "SPI" doesn't make the difference clear enough because SPI is not well understood (I just asked 3 developers and none of them gave me a difference between API and SPI that would have illuminated the distinction). I think the sentence added about choice of messaging technology is a better way to make that point.

(I just removed -spi from the section id in last commit too, hadn't noticed that before)